src/evaluation_system/api/plugin_manager.py

```diff
@@ -1056,6 +1056,53 @@ def get_history(
     )
 
 
+def get_history_all(
```

Comment:
For testifying on my machine i couldn't leave `user` as `None` to get the all users history. in case in operational, i mean in realty on the server if we can leave `user` as `None` and `User()` could get the `getUserDB ()`, then the function would be way easier without defining any new arg here:

```python
def get_history(
    plugin_name: Optional[str] = None,
    limit: int = -1,
    since: Optional[str] = None,
    until: Optional[str] = None,
    entry_ids: Optional[list[int]] = None,
    user: Optional[User] = None,
) -> QuerySet[History]:
    """Returns the history from the given user.

    This is just a wrapper for the defined db interface accessed via the
    user object. See :class:`evaluation_system.model.db.UserDB.getHistory`
    for more information on this interface.

    Parameters
    ----------
    plugin_name
        Name of plugin to get the history for
    limit
        Limits on number of results to get. -1 means no limit
    since
        Only get results after since.
    until
        Only get results earlier than until.
    entry_ids
        Result entry IDs to filter for.
    user :
        User to get plugin results for. If None, fetch results for all users.

    Returns
    -------
    QuerySet[History]
        Results from the database query.
    """
    user = user or User()
    
    return user.getUserDB().getHistory(
        tool_name = plugin_name.lower() if plugin_name else None,
        limit=limit,
        since=since,
        until=until,
        entry_ids=entry_ids,
        uid=user.getName() if user else None,
    )
```

Reply:
I have a very limited knowledge on the underlyings of these methods and I think you pose a valid concern, but I can tell you that in the web browser we already have the result-browser tab that is essentially doing that: as you can see [here](https://www.xces.dkrz.de/history/result-browser/) the (finished) jobs of every user in that freva-instance are visible to anyone.

Since the aim of Freva is (or at least was) to create a collaborative env within the researches of a project, and we allow (and encourage) people to share plugins (that is, one creates one so everyone else benefits on it), and data and have tools to even share the results of an output, I would say that at least from the ethical stand point that would not be a problem.

Reply:
Then great, but just let @antarcticrainforest have a look, maybe there is possibility to define `user` as `None` in `get_history ` func. because if we could implement the ideal idea, i mean [this one](https://github.com/FREVA-CLINT/freva/pull/198#discussion_r1613194575), then it would be way more efficient ...